fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/SocialMediaViewModel.java

```diff
@@ -68,11 +116,65 @@ public void openSend() {
     mOpenSendEvent.postValue(new SingleEvent<>(true));
   }
 
+  public void openSearch() {
+    mOpenSearchEvent.postValue(new SingleEvent<>(true));
+  }
+
   public void openFollowing() {
     mOpenFollowingEvent.postValue(new SingleEvent<>(true));
   }
 
   public void openProfile() {
     mOpenProfileEvent.postValue(new SingleEvent<>(true));
   }
+
+  public void sendNewChirpEvent() {
+    mSendNewChirpEvent.postValue(new SingleEvent<>(true));
+  }
+
+  public void setNumberCharsLeft(Integer numberChars) {
+    mNumberCharsLeft.setValue(numberChars);
+  }
+
+  public void setLaoId(String laoId) {
+    mLaoId.setValue(laoId);
+  }
+
+  /** Subscribe to a channel: /root/<lao id>/social/<sender> */
+  public void subscribeToChannel(String laoId) {
+    Log.d(TAG, "subscribing to channel: /root/" + laoId + "/social/<sender>");
+
+    try {
+      KeysetHandle publicKeysetHandle = mKeysetManager.getKeysetHandle().getPublicKeysetHandle();
+      String publicKey = Keys.getEncodedKey(publicKeysetHandle);
+      byte[] senderByte = Base64.getUrlDecoder().decode(publicKey);
+      String sender = new String(senderByte);
+      String channel = "/root/" + laoId + "/social/" + sender;
```

Comment:
Is sender supposed to be the public key ?
If not, the sender string won't be readable and will contains chars like "ו�_���b���'r�~�v��b3�y:j�" instead of something like "BdeV1F-ylAn0YgDvJ3LcfrB2jxbuiaC_DGIznnk6at8=" for the publicKey.
```suggestion
      String publicKey = Keys.getEncodedKey(publicKeysetHandle);
      String channel = "/root/" + laoId + "/social/" + publicKey;
```